### ISSUE TITLE
[Bug]: DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.

### ISSUE THREAD

**Issue by Jwiggiff** [remix-run/react-router#8834]
### What version of React Router are you using?

6.3.0

### Steps to Reproduce

1. Attempt to navigate to a different route using `useNavigate()` hook on form submission

### Expected Behavior

Redirect to new route (in this case a success page).

### Actual Behavior

URL changes in the address bar, however the page does not change and I get the following error in the console:
```
DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
```

If I reload the page afterwards, the route loads just fine.

**Comment by timdorr** [comment]
This doesn't sound like a bug in React Router. We don't have that code in our library (just the tests). It may be a bug with React or something in your app code.

**Comment by Jwiggiff** [comment]
I have resolved the issue! Yes you are correct, it is not an issue with react router that is my bad. I'll leave the solution here in case anyone makes their way to this post with the same problem:

It has to do with using react fragments. The root component of the main route was the App component. However, in the App component I had a fragment with two components inside. When navigating, it attempted to remove the App component which did not actually exist since it was a fragment which does not add to the actual dom.

TL;DR: replace the fragment with a `div` instead.

**Comment by arananegra** [comment]
@Jwiggiff god you are my savior. I've been experiencing this for a couple of weeks (I don't know if upgrading to React 18 has to do with it), and it was painful to deal with. Changing fragment to div in the "main" App component made it

**Comment by arrickx** [comment]
@Jwiggiff I changed all the fragment with a div. now it's working. thanks!!

**Comment by serge1000** [comment]
This helped me too. Thank you @Jwiggiff !!!

**Comment by kaiseradr** [comment]
Thank you @Jwiggiff! This a big help, I was struggling with the same issue. <3

**Comment by tizah** [comment]
exactly what iI needed to do...thanks @Jwiggiff

**Comment by luanfonsecap** [comment]
Thanks a lot @Jwiggiff, you made my day!

**Comment by zakiradel** [comment]
thanks man really save my time

**Comment by soulcm** [comment]
I have the same error, I will try it to fix

**Comment by kishanup78910** [comment]
but issue not resolved

**Comment by orochi235** [comment]
Just ran into this myself; it was a similar issue where I'd tried to replace a plain `<div>` near the root of the app with a fragment, because the div seemed like overkill. It wasn't, for reasons @Jwiggiff detailed above. It caused multiple copies of the app to render at once, in divergent states, leading me to wonder how the hell both sides of a ternary were rendering at once, among other things.

A better error message would have made this a lot easier to track down. Is there any way react-router could at least raise a warning if you pass it a component with fragments in dangerous places?

**Comment by CalvinJamesHeath** [comment]
Did not work for me. I have no fragments and still have this issue.

**Comment by felixhagspiel** [comment]
@CalvinJamesHeath for me the issue was not using `next/script`-script tags for things like google analytics and other external scripts.

**Comment by antonmorya** [comment]
Hey 
Theoretically, will this problem be solved if my `Fragment` has a `key`? Or `removeChild` work with real DOM nodes?

**Comment by rishianand39** [comment]
Just add key props (add value which change when you navigate) to the current page wrapper. When you try to change you path name whichout going to differnt page, page need to re-render. If you do not pass key it partially render the page and that's why we see that error.

**Comment by danneu** [comment]
I hope this isn't inappropriate/noisy for the maintainers, but this is the first google result for the general react error, and I have to thank @Jwiggiff for pointing out their solution.

I went down every rabbit hole, yet swapping `<>` for `<div>` in my App.tsx fixed it.

It's humbling because even after 10+ years of React, I doubt I would have considered this approach if given a whole month to look for it in a large production codebase.

**Comment by daoquangphuong** [comment]
If you put something like this: `<div> {isEnable ? "Enable" : "Disabled"} <span>this function</span> </div>` then you will get the error with google translate.
So to avoid it, make sure you never put a text node with a tag node at same level.
so you can fix like this: 
`<div> {isEnable ? "Enable" : "Disabled"} this function </div>`
or
`<div> <span>{isEnable ? "Enable" : "Disabled"}</span> <span>this function</span> </div>`

**Comment by robvanderleek** [comment]
FWIW, I bumped into this error message with React and FabricJS 7, where one of the components returned a single `<canvas ...>` element that is managed by Fabric.
Adding a `<div>` around the `canvas` made the problem go away.
